lib/Conversion/AffineToPipeline/AffineToPipeline.cpp

```diff
@@ -57,20 +57,73 @@ struct AffineToPipeline : public AffineToPipelineBase<AffineToPipeline> {
   LogicalResult populateOperatorTypes(SmallVectorImpl<AffineForOp> &loopNest);
   LogicalResult solveSchedulingProblem(SmallVectorImpl<AffineForOp> &loopNest);
   LogicalResult createPipelinePipeline(SmallVectorImpl<AffineForOp> &loopNest);
+  LogicalResult constantCSE();
+  LogicalResult unrollAndForwardStores();
 
   CyclicSchedulingAnalysis *schedulingAnalysis;
 };
 
+static LogicalResult
+forwardStoreToLoad(AffineReadOpInterface loadOp,
+                   SmallVectorImpl<Operation *> &opsToErase) {
+
+  Operation *prevOp = loadOp->getPrevNode();
+  while (prevOp) {
+    if (prevOp->getParentRegion() != loadOp->getParentRegion())
+      return failure();
```

Comment:
Isn't this check redudandant? getPrevOp should give things in the same block. I'm not sure how they can move across regions.